website/addons/dataverse/static/dataverseNodeConfig.js

```diff
@@ -199,25 +239,37 @@ ViewModel.prototype.updateFromData = function(data) {
     self.nodeHasAuth(data.nodeHasAuth);
     self.userHasAuth(data.userHasAuth);
     self.userIsOwner(data.userIsOwner);
+    self.hosts(data.hosts);
 
     if (self.nodeHasAuth()) {
+        self.savedHost(data.dataverseHost);
         self.dataverses(data.dataverses);
         self.savedDataverseAlias(data.savedDataverse.alias);
         self.savedDataverseTitle(data.savedDataverse.title);
         self.selectedDataverseAlias(data.savedDataverse.alias);
         self.savedDatasetDoi(data.savedDataset.doi);
         self.savedDatasetTitle(data.savedDataset.title);
-        self.connected(data.connected);
+        self.validCredentials(data.connected);
         if (self.userIsOwner()) {
             self.getDatasets(); // Sets datasets, selectedDatasetDoi
         }
     }
 };
 
+/** Reset all fields from Dataverse host selection modal */
+ViewModel.prototype.clearModal = function() {
+    var self = this;
+    self.message('');
+    self.messageClass('text-info');
+    self.apiToken(undefined);
```

Comment:
`null` is more appropriate for "blank" values. `undefined` is used for only for undefined/undeclared variables.